### PR TITLE
TW-1492: Fix the message blink when received new message in chat

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket:

- #1492 

### NEED MERGE:

- https://github.com/linagora/inview_notifier_list/pull/1

### Resolved:

https://github.com/linagora/twake-on-matrix/assets/99852347/8e6d98b2-46ca-4e1f-b1e8-faac5a3133b4

https://github.com/linagora/twake-on-matrix/assets/99852347/4c1b6fbc-e9c1-4586-92aa-bcc92416e8a2

### CODE REVIEW

**Review by Te-Z — APPROVED** [review]

File: lib/pages/chat/events/message/message.dart [hunk 1]
```diff
@@ -78,13 +79,61 @@ class Message extends StatelessWidget {
 
   static final responsiveUtils = getIt.get<ResponsiveUtils>();
 
+  @override
+  State<Message> createState() => _MessageState();
+}
+
+class _MessageState extends State<Message> {
+  late InViewState inViewState;
```
Comment (sherlockvn) [inline lib/pages/chat/events/message/message.dart:87]: late ?

File: lib/pages/chat/events/message/message.dart [hunk 2]
```diff
@@ -78,13 +79,61 @@ class Message extends StatelessWidget {
...
+
+  final inviewNotifier = ValueNotifier(false);
+
+  @override
+  void initState() {
+    super.initState();
+    inViewState = InViewNotifierListCustom.of(context)!;
```
Comment (sherlockvn) [inline lib/pages/chat/events/message/message.dart:94]: may not have the context at this time yet

Comment (sherlockvn) [reply lib/pages/chat/events/message/message.dart:94]: why don't you pass down the `inViewState` from ChatEventList class

File: lib/pages/chat/events/message/message.dart [hunk 3]
```diff
@@ -78,13 +79,61 @@ class Message extends StatelessWidget {
...
+    inViewState.addListener(_inviewStateListener);
+    inViewState.addContext(context: context, id: widget.event.eventId);
```
Comment (sherlockvn) [inline lib/pages/chat/events/message/message.dart:96]: same

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**